Tensile/Components/LraTileAssignment.py

```diff
@@ -138,7 +144,59 @@ def __call__(self, writer, kernel, tP):
             "0. thread id in wave: wtid = tid %% wavelength(%u)" % waveWidth)
         kStr += vectorStaticRemainder(dummy, tReg, kReg, kernel["MatrixInstN"], tmpVgpr, tmpSgpr, \
             "1. N offset: nIdx = wtid %% MI_N(%u)" % kernel["MatrixInstN"])
-        kStr += staticMultiply(vgpr(tReg), vgpr(tReg), strideTile, sgpr(tmpSgpr), \
+        if kernel["ThreadSeparateGlobalRead%c"%tc]:
+          #Nidx offset calculation
+          # each load fetches tuple<K1,Nidx,K0> mapped to wavefront load tuple<TSGR<<1,wavefront/depthU/TSGR<<1, depth//TSGR<<1) 
+          kStr += vectorStaticDivide(mReg, tReg, NblockSizePerLoad, tmpVgpr, tmpSgpr, \
+              "1. N offset: nIdx_upper = nIdx >> NblockSizePerLoad(%u)" % NblockSizePerLoad)
+          kStr += staticMultiply(vgpr(mReg), vgpr(mReg), NblockSizePerLoad*kernel["_DepthULds"], sgpr(tmpSgpr), \
+              "1. N offset: nIdx_upper_offset = nIdx_upper * nStride(%u)" % (NblockSizePerLoad*kernel["_DepthULds"]))
+          KelementsPerMFrag = kernel["_DepthULds"]//(kernel["ThreadSeparateGlobalRead%c"%tc]*2)
+          KlanesPerMFrag = KelementsPerMFrag // kernel["GlobalLoadVectorWidth%c"%tc] 
+          MidxScale = KelementsPerMFrag
+          MidxRemainder = NblockSizePerLoad
+          if ((KlanesPerMFrag == 8 and (kernel["GlobalLoadVectorWidth%c"%tc] * tP["bpe"]) == 8) or 
+              (KlanesPerMFrag == 4 and (kernel["GlobalLoadVectorWidth%c"%tc] * tP["bpe"]) == 16)):
+            kStr += inst("v_and_b32",vgpr(mReg1),(MidxRemainder-1),vgpr(tReg), \
+              "1. N offset: nIdx_load = nIdx %% NblockSizePerLoad(%u)" % NblockSizePerLoad)
+            kStr += staticMultiply(vgpr(mReg1), vgpr(mReg1), (kernel["GlobalLoadVectorWidth%c"%tc]*KlanesPerMFrag), sgpr(tmpSgpr), \
+              "1. N offset: nIdx_load = nIdx_load * nStride(%u)" % strideTile)
+          elif ((KlanesPerMFrag == 4 and (kernel["GlobalLoadVectorWidth%c"%tc] * tP["bpe"]) == 8) or
+                (KlanesPerMFrag == 2 and (kernel["GlobalLoadVectorWidth%c"%tc] * tP["bpe"]) == 8)):
+            numMidxPer8Ldslanes =  8 // KlanesPerMFrag
+            MidxScale = KlanesPerMFrag * 4 // tP["bpe"]
+            kStr += inst("v_and_b32",vgpr(tReg),(MidxRemainder-1),vgpr(tReg), \
+              "1. N offset: nIdx_load = nIdx %% NblockSizePerLoad(%u)" % NblockSizePerLoad)
+            kStr += inst("v_and_b32",vgpr(mReg1),(numMidxPer8Ldslanes-1),vgpr(tReg), \
+              "1. N offset: nIdx_lower = mIdx %% numMidxPer8Ldslanes(%u)" % numMidxPer8Ldslanes)
+            kStr += staticMultiply(vgpr(mReg1), vgpr(mReg1), MidxScale, sgpr(tmpSgpr), \
+              "1. N offset: nIdx_lower_offset = nIdx_lower * nStride(%u)" % MidxScale)
+            kStr += vectorStaticDivide(mReg2, tReg, numMidxPer8Ldslanes, tmpVgpr, tmpSgpr, \
+            "1. N offset: nIdx_load = wtid %% nStride(%u)" % numMidxPer8Ldslanes)
```

Comment:
It looks like the comment here is not accurate.
It is a division, but the comment says % (=mod).